pyvo/mivot/utils/vocabulary.py

```diff
@@ -0,0 +1,155 @@
+"""
+MIVOT vocabulary and regular expressions.
+"""
+import re
+from astropy import units as u
+from pyvo.utils import prototype_feature
+
+
+class Constant:
+    """
+    Class used to set constant to identify XML attributes added to the MIVOT ATTRIBUTES
+    """
+    FIRST_TABLE = "first_table"
+    FIELD_UNIT = "field_unit"
+    COL_INDEX = "col_index"
+    ROOT_COLLECTION = "root_collection"
+    NOT_SET = "NotSet"
+    ANONYMOUS_TABLE = "AnonymousTable"
+
+
+# Regexp pattern to check no valid mapping is present
+NoMapping = re.compile(r".REPORT\s+status=['\"]KO")
+
+unit_mapping = {
+    "deg": u.degree,
+    "rad": u.radian,
+    "hourangle": u.hourangle,
+    "arcsec": u.arcsec,
+    "mas": u.mas,
+    "pc": u.pc,
+    "km": u.km,
+    "m": u.m,
+    "mas/yr": u.mas / u.yr,
+    "mas/y": u.mas / u.yr,
+    "km/s": u.km / u.s,
+}
+
+
+regex_patterns = {'iso': r'^(-?(?:[1-9][0-9]*)?[0-9]{4})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12]'
+                         r'[0-9]) (2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])(\.[0-9]+)?'
+                         r'(Z|[+-](?:2[0-3]|[01][0-9]):[0-5][0-9])?$',
+                  'isot': r'^(-?(?:[1-9][0-9]*)?[0-9]{4})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])'
+                          r'T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])(\.[0-9]+)?(Z|[+-]'
+                          r'(?:2[0-3]|[01][0-9]):[0-5][0-9])?$',
+                  'yday': r'^(\d{4}):(\d{3}):(\d{2}):(\d{2}):(\d{2}\.\d+)$',
+                  'ymdhms': r"^\{'year': (\d+), 'month': (\d+), 'day': (\d+)"
+                            r"(?:, 'hour': (\d+))?(?:, 'minute': (\d+))?(?:, 'second': (\d+))?\}$"}
+
+
+regex_format = {'iso': re.compile(regex_patterns['iso']),
+                'isot': re.compile(regex_patterns['isot']),
+                'yday': re.compile(regex_patterns['yday']),
+                'ymdhms': re.compile(regex_patterns['ymdhms'])}
+
+
+@prototype_feature('MIVOT')
+class Ele:
+    """
+    Constant used to identify MIVOT Element
+    """
+    namespace = ""
```

Comment:
Is this empty `namespace` a sort of placeholder for future features?  It doesn't seem needed or used now.

Reply:
Unused code removed from vocabulary.py (c24931f782c1092253511548b506d5f3abf7bd78).